queries.sql

```diff
@@ -15,3 +15,44 @@ SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name NOT LIKE 'Gabumon';
 
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN;
+ SAVEPOINT sp1;
+ UPDATE animals
+ SET species = 'unspecified';
+ ROLLBACK TO sp1;
+COMMIT;
+
+BEGIN;
+UPDATE animals
+SET species = CASE WHEN name LIKE '%mon' THEN 'digimon' ELSE 'pokemon' END;
+COMMIT;
+
+BEGIN;
+SAVEPOINT sp1;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK TO sp1;
+SELECT * FROM animals;
+COMMIT;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT sp1;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO sp1;
+UPDATE animals SET weight_kg = CASE WHEN weight_kg < 0 THEN weight_kg * -1 ELSE weight_kg END;
+COMMIT;
+
+
+SELECT COUNT(name) AS number_of_animals FROM animals;
+
+SELECT COUNT(name) AS tried_to_escape FROM animals WHERE escape_attempts > 0;
```

Comment:
@topeogunleye Thank you for the correction, I must admit that i have learnt a lot from your review, But repeating some lines in every comments and my correct code in your comments made a little confusion and it is complected to find the exact problems and suggested changes. .

Data Updated accordingly.

Reply:
@topeogunleye Thank you for the correction, I must admit that i have learnt a lot from your review, But repeating some lines in every comments and my correct code in your comments made a little confusion and it is complected to find the exact problems and suggested changes. .

Code updated accordingly.

Reply:
@topeogunleye 
I want to express my gratitude for taking the time to review my code. Your corrections and suggestions have been immensely valuable, and I have learned a lot from your review.

However, I would like to offer a suggestion to improve the clarity of the feedback. I noticed that some lines in your comments, including my corrected code, are repeated across different sections. As a result, it made it a little confusing for me to identify the exact problems and the corresponding suggested changes.

In the future, if it's possible, could you consider consolidating similar comments or using references to specific sections of the code to provide more focused feedback? This would greatly help in understanding and addressing the issues efficiently.

Once again, I truly appreciate your effort and expertise in reviewing my code. Your feedback has been instrumental in my growth as a developer.

Thanking You
Nahid Raihan